### PR TITLE
added check for 0 index item on optionItem

### PR DESCRIPTION
**Pull request by moeabm**
added check for 0 index item on optionItem for mouseover and mouseup
https://github.com/cibernox/ember-power-select/issues/402

### CODE REVIEW

File: addon/components/power-select/options.js [hunk 1]
```diff
@@ -15,12 +15,12 @@ export default Ember.Component.extend({
...
     this.element.addEventListener('mouseup', e => {
       const optionItem = Ember.$(e.target).closest('[data-option-index]');
-      if (!optionItem) { return; }
+      if (!optionItem || !(0 in optionItem)) { return; }
       this.get('select.actions.choose')(this._optionFromIndex(optionItem[0].dataset.optionIndex), e);
     });
     this.element.addEventListener('mouseover', e => {
       const optionItem = Ember.$(e.target).closest('[data-option-index]');
-      if (!optionItem) { return; }
+      if (!optionItem || !(0 in optionItem)) { return; }
```
Comment (cibernox) [inline addon/components/power-select/options.js]: I've never seen this construction for checking if a value is zero. Is there any difference with the more common `!optionItem && optionItem !== 0`? If there is none, I'd say that I prefer the `!==` one, for familiarity.

**Comment by AndrewMoore10** [comment]
I suppose `!optionItem[0]` would be equivalent.

File: addon/components/power-select/options.js (hunk 1 above)
Comment (AndrewMoore10) [inline addon/components/power-select/options.js]: Checking for existence of the zero index element. I suppose !optionItem[0] would be equivalent.

**Comment by cibernox** [comment]
Thanks!

**Pull request merged**